plugins/destination/bigquery/client/write.go

```diff
@@ -95,8 +95,18 @@ func (c *Client) Write(ctx context.Context, tables schema.Tables, res <-chan *pl
 		})
 	}
 
-	for r := range res {
-		workers[r.TableName].writeChan <- r.Data
+	done := false
+	for !done {
```

Comment:
```suggestion
	for more := true; more; {
		select {
		case r, ok := <-res:
			if !ok {
				more = false
				break
			}
			workers[r.TableName].writeChan <- r.Data
		case <-gctx.Done():
			more = false
		}
	}
```
Better not do extra flip

Reply:
`more` is not really accurate here though - we're exiting because something went wrong, not because there aren't any more entries